src/LuaTransformer.ts

```diff
@@ -2628,9 +2628,20 @@ export class LuaTransformer {
     }
 
     public transformForInStatement(statement: ts.ForInStatement): StatementVisitResult {
-        // Get variable identifier
-        const variable = (statement.initializer as ts.VariableDeclarationList).declarations[0];
-        const identifier = variable.name as ts.Identifier;
+        // Transform iteration variable
+        let iterationVariable: tstl.Identifier;
+        if (
+            ts.isVariableDeclarationList(statement.initializer) &&
+            !ts.isArrayBindingPattern(statement.initializer.declarations[0].name) &&
```

Comment:
Wouldn't it be cleaner to include the only one expected node kind instead of filtering out unexpected ones?
```ts
ts.isIdentifier(statement.initializer.declarations[0].name)
````